SofaKernel/modules/SofaSimpleFem/src/SofaSimpleFem/TetrahedronFEMForceFieldImplPolar.h

```diff
@@ -0,0 +1,62 @@
+*                                                                             *
+* This program is free software; you can redistribute it and/or modify it     *
+* under the terms of the GNU Lesser General Public License as published by    *
+* the Free Software Foundation; either version 2.1 of the License, or (at     *
+* your option) any later version.                                             *
+*                                                                             *
+* This program is distributed in the hope that it will be useful, but WITHOUT *
+* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or       *
+* FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License *
+* for more details.                                                           *
+*                                                                             *
+* You should have received a copy of the GNU Lesser General Public License    *
+* along with this program. If not, see <http://www.gnu.org/licenses/>.        *
+*******************************************************************************
+* Authors: The SOFA Team and external contributors (see Authors.txt)          *
+*                                                                             *
+* Contact information: contact@sofa-framework.org                             *
+******************************************************************************/
+#pragma once
+
+#include <SofaSimpleFem/TetrahedronFEMForceFieldImplCorotational.h>
+
+namespace sofa::component::forcefield
+{
+
+template<class DataTypes>
+class TetrahedronFEMForceFieldImplPolar : public TetrahedronFEMForceFieldImplCorotational<DataTypes>
+{
+protected:
+
+    using Inherit = TetrahedronFEMForceFieldImplCorotational<DataTypes>;
+    using Inherit::Transformation;
+    using Inherit::Coord;
+    using Inherit::VecCoord;
+    using Inherit::Real;
+    using Inherit::Displacement;
+    using Inherit::Element;
+    using Inherit::FiniteElementArrays;
+
+    void computeRotation(Transformation& rotation, const VecCoord& positions, Index a, Index b, Index c, Index d, unsigned elementIndex) const override;
+
+    void computeDisplacement(Displacement& displacement, const type::fixed_array<type::VecNoInit<3, Real>,4>& deformedElement,
+            const type::fixed_array<Coord,4>& rotatedInitialElement) const override;
+
+    type::fixed_array<Coord,4>
+    computeRotatedInitialElement(const Transformation& rotation, const Element& element, const VecCoord& initialPoints) const override;
+
+    void computeDeformedElement(type::fixed_array<type::VecNoInit<3, Real>,4>& deforme,
+        const Transformation& rotation, const Coord& xa, const Coord& xb, const Coord& xc, const Coord& xd) const override;
+
+    void addForceAssembled(const FiniteElementArrays& finiteElementArrays) override;
+
+};
+
+
+
+}
+
+#include <SofaSimpleFem/TetrahedronFEMForceFieldImplPolar.inl>
```

Comment:
This is an un-usual pattern. 

In general the .inl in included where some template instanciation happens and not in the .h. 
(Actually, I suspect that using a .inl exists specifically to reduce the .h to the bare minimum, thus reducing the include graph. )

Reply:
I agree with your comment but only in the scope of SOFA. It's not so unusual to include .inl files in the header outside of SOFA. See also my other comment.